x/stakeibc/keeper/validator_selection.go

```diff
@@ -244,26 +249,33 @@ func (k Keeper) GetValidatorDelegationDifferences(ctx sdk.Context, hostZone type
 	return delegationDeltas, nil
 }
 
-// This will get the target validator delegation for the given hostZone
-// such that the total validator delegation is equal to the finalDelegation
-// output key is ADDRESS not NAME
-func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, finalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
+// This will split a total delegation amount across validators, according to weights
+// It returns a map of each portion, key'd on validator address
+// Validator's with a slash query in progres are excluded
+func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, totalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
 	// Confirm the expected delegation amount is greater than 0
-	if !finalDelegation.IsPositive() {
+	if !totalDelegation.IsPositive() {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest,
-			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", finalDelegation)
+			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", totalDelegation)
+	}
+
+	// Ignore any validators with a slash query in progress
+	validators := []types.Validator{}
+	for _, validator := range hostZone.Validators {
+		if !validator.SlashQueryInProgress {
```

Comment:
Still thinking through is this really what we want to do, by doing the ICQ check deep inside this function and filtering out validators instead of at the time of specific ICA kick off could potentially affect all ICAs?  In the past I thought we had said we wouldn't ever want to block ICAs needed for standard route liquid stakes by the ICQ but we would want to block LSM liquid stakes (since we need an accurate exchange rate to deal with the shares representation).

Is there a guarantee that these helpers are only going to be called in the LSM related routes but not the pre-existing rebalance type functions?  Also, by filtering out validators, the balancing logic is going to be slightly different.  Instead of computing how much should go to that validator but then waiting to execute it until ICQs are clear, any mass which would have gone to that validator will now be rebalanced over to other validators.

Reply:
>  In the past I thought we had said we wouldn't ever want to block ICAs needed for standard route liquid stakes by the ICQ but we would want to block LSM liquid stakes

This is true, but we decided against it eventually considering how dangerous it is to have conflicts and the fact that it only affect ICAs _to that validator_. The case we want to avoid is something like skipping a full epochly delegation.

Re: rebalancing, this is something I realized late last night too. I think we have to subtract out the validators that have a slash query in progress when we get the total

Reply:
Yes I think this is critical or any validator with slash query in progress will lose all their delegation in a rebalance!  Just saw the update and it looks good, just avoiding those validators with slash query in progress from the total since they will be avoided in the rebalance logic.

Reply:
From our offline conversation about this, I think we need to apply the same exclusion logic on the weights when computing the TargetValAmounts so that only the relative weights of validators which do not have a slash query in progress are considered in that computation.

Reply:
aren't they already excluded [here](https://github.com/Stride-Labs/stride/pull/844/files#diff-7eb0c0f79f6e24ba948f073cca1f9f6848f614e735f052b4714b82657237ecebR280)?

Reply:
Yes sorry, I see we build the list of validators and then pass it as the incoming argument so any with ICQ active are already excluded.  Then I think this issue is resolved.